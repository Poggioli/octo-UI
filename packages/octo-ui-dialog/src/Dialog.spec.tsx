import { render } from "@testing-library/react";
import {
  Dialog,
  DialogClose,
  DialogCloseIcon,
  DialogContent,
  DialogDescription,
  DialogTitle,
  DialogTrigger,
} from "./Dialog";

describe("<Dialog />", () => {
  it(`GIVEN a <Dialog />
      WHEN render is called
      THEN should render a <Dialog />`, () => {
    const rendered = render(
      <Dialog defaultOpen>
        <DialogTrigger>
          <button>Trigger</button>
        </DialogTrigger>
        <DialogContent blur>
          <DialogCloseIcon />
          <DialogTitle>Edit profile</DialogTitle>
          <DialogDescription>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ut
            quam in dui facilisis consequat. Ut quis varius neque. Nulla et
            tellus velit.
          </DialogDescription>

          <DialogClose>
            <button>Cancel</button>
          </DialogClose>
          <DialogClose>
            <button>Save</button>
          </DialogClose>
        </DialogContent>
      </Dialog>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Dialog />
      WHEN render is called
      THEN should render a <Dialog />`, () => {
    const rendered = render(
      <Dialog>
        <DialogTrigger>
          <button>Trigger</button>
        </DialogTrigger>
        <DialogContent blur>
          <DialogCloseIcon />
          <DialogTitle>Edit profile</DialogTitle>
          <DialogDescription>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ut
            quam in dui facilisis consequat. Ut quis varius neque. Nulla et
            tellus velit.
          </DialogDescription>

          <DialogClose>
            <button>Cancel</button>
          </DialogClose>
          <DialogClose>
            <button>Save</button>
          </DialogClose>
        </DialogContent>
      </Dialog>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Dialog />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Dialog.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <DialogClose />
      WHEN toString is called
      THEN return a string`, () => {
    expect(DialogClose.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <DialogCloseIcon />
      WHEN toString is called
      THEN return a string`, () => {
    expect(DialogCloseIcon.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <DialogContent />
      WHEN toString is called
      THEN return a string`, () => {
    expect(DialogContent.toString()).toBe(".octo-UI-c-hEFdbp");
  });

  it(`GIVEN a <DialogDescription />
      WHEN toString is called
      THEN return a string`, () => {
    expect(DialogDescription.toString()).toBe(".octo-UI-c-ceIQAL");
  });

  it(`GIVEN a <DialogTitle />
      WHEN toString is called
      THEN return a string`, () => {
    expect(DialogTitle.toString()).toBe(".octo-UI-c-dAnBzO");
  });

  it(`GIVEN a <DialogTrigger />
      WHEN toString is called
      THEN return a string`, () => {
    expect(DialogTrigger.toString()).toBe(".octo-UI-c-fDiieJ");
  });
});
