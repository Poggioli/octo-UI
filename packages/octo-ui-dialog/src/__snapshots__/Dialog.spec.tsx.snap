// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> GIVEN a <Dialog />
      WHEN render is called
      THEN should render a <Dialog /> 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="octo-UI-c-fDiieJ"
    data-state="open"
    type="button"
  >
    Trigger
  </button>
</div>
`;

exports[`<Dialog /> GIVEN a <Dialog />
      WHEN render is called
      THEN should render a <Dialog /> 2`] = `
<div>
  <button
    aria-controls="radix-:r3:"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="octo-UI-c-fDiieJ"
    data-state="closed"
    type="button"
  >
    Trigger
  </button>
</div>
`;
